lib/mirage.ml

```diff
@@ -2100,6 +2234,17 @@ let configure_main_xe t =
   append oc "VBD=$(xe vbd-create vm-uuid=$VM vdi-uuid=$VDI device=0)";
   append oc "xe vbd-param-set uuid=$VBD bootable=true";
   append oc "xe vbd-param-set uuid=$VBD other-config:owner=true";
+  List.iter (fun b ->
+    append oc "echo Uploading data VDI %s" b.Block.filename;
+    append oc "echo VDI=$VDI";
+    append oc "SIZE=$(stat --format '%%s' %s/%s)" t.root b.Block.filename;
+    append oc "POOL=$(xe pool-list params=uuid --minimal)";
+    append oc "SR=$(xe pool-list uuid=$POOL params=default-SR --minimal)";
+    append oc "VDI=$(xe vdi-create type=user name-label='%s' virtual-size=$SIZE sr-uuid=$SR)" b.Block.filename;
+    append oc "xe vdi-import uuid=$VDI filename=%s/%s" t.root b.Block.filename;
+    append oc "VBD=$(xe vbd-create vm-uuid=$VM vdi-uuid=$VDI device=%d)" b.Block.number;
+    append oc "xe vbd-param-set uuid=$VBD other-config:owner=true";
+  ) (Hashtbl.fold (fun _ v acc -> v :: acc) all_blocks []);
```

Comment:
Have a look up at line 2191 -- that's where the xl config file is built. The main semantic difference is that `xe` uses the remote interface and uploads a copy of the disk, whereas the `xl` config file uses the disk in-place.